test/test_ssl.py

```diff
@@ -91,71 +92,38 @@ def test_no_ssl_module(self):
 
         # Implied
         self.assertRaises(ConfigurationError,
-                          MongoClient, ssl_certfile=CLIENT_PEM)
+                          MongoClient, tlsCertificateKeyFile=CLIENT_PEM)
 
     @unittest.skipUnless(HAVE_SSL, "The ssl module is not available.")
     @ignore_deprecations
     def test_config_ssl(self):
         # Tests various ssl configurations
         self.assertRaises(ValueError, MongoClient, ssl='foo')
-        self.assertRaises(ConfigurationError,
```

Comment:
The spec does not mandate that ssl/tls=False + one of the tls* options implying a conflicting value of ssl/tls should error. It mandates only that multiple conflicting values of ssl/tls should error and we do adhere to that. Consequently, setting an option like tlsCAFile implicitly sets tls=True and a conflicting tls=False in the URI is simply ignored. 
@ShaneHarvey what is your opinion here? Do you think we should error? I think it is not required since we seem to do the more secure thing and since its 4.0 a breaking change is fine.

Reply:
I think it makes more since for us to error on things like this:
```python
MongoClient('mongodb://host/?tls=false&tlsCAFile=<...>')  # Conflict
MongoClient('mongodb://host/?tls=false&tlsInsecure=false')  # Conflict
```

Reply:
Agreed... Reinstated this behavior.